src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,73 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final Pair[] elements;
+    private int size;
+
+    public StorageImpl() {
+        elements = new Pair[MAX_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        if (size == MAX_SIZE) {
+            throw new ArrayIndexOutOfBoundsException("Array already full");
+        }
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i] == null) {
+                elements[size] = new Pair<>(key, value);
+                size++;
+                break;
+            } else if (elements[i].getKey() == key
+                    || elements[i].getKey() != null && elements[i].getKey().equals(key)) {
+                elements[i].setValue(value);
+                break;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < elements.length; i++) {
+            if (elements[i] == null) {
+                return null;
+            } else if (elements[i].getKey() == null && key == null
+                    || elements[i].getKey() != null && elements[i].getKey().equals(key)) {
+                return (V)elements[i].getValue();
+            }
```

Comment:
1. fix prev mentors comments before requesting next PR review.
![image](https://github.com/mate-academy/jv-storage/assets/73041091/6105b587-5c6f-4759-a042-44712c6c18a8)
2. any reason to add explicit cast here?
```suggestion
            if (key == elements[i].key || key != null && key.equals(elements[i].key)) {
                return elements[i].getValue();
            }
```

Reply:
There is a compile error without cast